File: src/main/java/com/govtech/assignment/service/UserService.java
package com.govtech.assignment.service;

import java.util.Set;

import com.govtech.assignment.entity.User;
import com.govtech.assignment.request.CreateUserRequest;
import com.govtech.assignment.response.CreateUserResponse;
import com.govtech.assignment.response.UserResponse;
import com.govtech.assignment.response.UsersResponse;

public interface UserService {

	CreateUserResponse create(CreateUserRequest request);

	User findById(String id);

	UserResponse getByAccountId(String accountId);

	UserResponse getById(String id);

	UsersResponse find(Integer pageOffset, Integer pageLimit, String searchText, String sortBy, String sortOrder);

	Set<User> findByIds(Set<String> userIds);

}

File: src/main/frontend/govtech-assignment-ui/src/app/providers/user/user.service.ts
import { Injectable } from '@angular/core';
import { RestClientService } from '../rest-client/rest-client.service';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private baseUrl: string;

  constructor(private restClientService: RestClientService) { 
    this.baseUrl = 'api/v1/user';
  }

  getByAccountId(accountId:string) {
    return this.restClientService.get(this.baseUrl+"/account/"+accountId);
  }

  getById(id:string) {
    return this.restClientService.get(this.baseUrl+"/"+id);
  }
  
  create(createUserRequest:any){
    return this.restClientService.post(createUserRequest,this.baseUrl);
  }

  find(pageOffset:number,pageLimit:number,searchText:string,sortBy:string,sortOrder:string) {
    searchText=searchText=="" || !searchText?"ALL":searchText;
    return this.restClientService.get(this.baseUrl+"/"+pageOffset+"/"+pageLimit+"/"+searchText+"/"+sortBy+"/"+sortOrder);
  }

}
